cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/BrokerSetAwareGoal.java

```diff
@@ -214,7 +216,7 @@ protected void rebalanceForBroker(Broker broker, ClusterModel clusterModel, Set<
                                     OptimizationOptions optimizationOptions) throws OptimizationFailureException {
     LOG.debug("balancing broker {}, optimized goals = {}", broker, optimizedGoals);
 
-    String currentBrokerSetId = _brokerSetResolutionHelper.brokerSetId(broker);
+    String currentBrokerSetId = _brokerSetResolutionHelper.brokerSetId(broker.id());
```

Comment:
Hi Mohit, sorry I didn't get it. With the change in this PR, we change from using broker object to broker id, not the opposite way.

As for the case that broker is null, we never seen such case before. I think in that case the goal will fail. I'm fine with that because it should success in the next run once the broker object go back to non-null.

(I checked some other places in cruise-control, and didn't see any places make the assumption that broker object can be null. So I think it should be fine?)

Reply:
Please Ignore my comment for broker.id()

Broker object becomes null when the broker is provisioned and is being set-up. While populating rackIds to brokers the metadata sometimes returned a broker information without rack id and id rack id is not available then broker object comes as null. I think this is an edge case and unless reproduced again, you can continue with this PR.